File: 56-merge-intervals/56-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
      int n = intervals.size();
      sort(intervals.begin(), intervals.end());
      // sort(intervals.begin(), intervals.end(), [](const vector<int> &a, const vector<int>  &b) -> bool{return a[0] <= b[0];});
      vector<vector<int>> ans;
      vector<int>cur{intervals[0][0], intervals[0][1]};
      for(int i=1;i<n;i++)
      {
        if(intervals[i][0] <= cur[1])
        {
          //overlap
          cur[1] = max(cur[1], intervals[i][1]);
        }
        else
        {
          ans.push_back(cur);
          cur[0] = intervals[i][0];
          cur[1] = intervals[i][1]; 
        }
      }
      ans.push_back(cur);
      return ans;
    }
};

File: 56-merge-intervals/56-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        ans = []
        intervals.sort(key=lambda x:x[0])
        start,end = intervals[0][0],intervals[0][1]
        for i in range(1,len(intervals)):
          if intervals[i][0] <= end:
            end = max(end, intervals[i][1])
          else:
            ans.append([start,end])
            start = intervals[i][0]
            end = intervals[i][1]
        ans.append([start,end])
        return ans